### PR TITLE
[Term Entry] PyTorch GPU Acceleration with CUDA: Performance Optimization

### LINKED ISSUE #5965: [Term Entry] PyTorch GPU Acceleration with CUDA: Performance Optimization
### Reason/inspiration (optional)

We would like a new term entry on the `Performance Optimization` concept under Pytorch. The entry should go in a new file under `docs/content/pytorch/concepts/gpu-acceleration-with-cuda/terms/performance-optimization/performance-optimization.md`.

The entry should include:

- An introduction to the concept
- A `Syntax` section that provides the syntax for the concept
- An `Example` section that provides an example demonstrating the concept in use

Please refer to the [term entry template](https://github.com/Codecademy/docs/blob/main/documentation/term-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md) and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by IK-MEE**
### Description
Added a new Term Entry for PyTorch GPU Acceleration with CUDA: Performance Optimization

### Issue Solved
Closes #5965

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [ ] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

**Comment by IK-MEE** [comment]
Any edits or add-on suggestions are all welcome! 😄

**Review by Sriparno08 — APPROVED** [review]
Looks good for a second review, @IK-MEE!

File: content/pytorch/concepts/gpu-acceleration-with-cuda/terms/performance-optimization/performance-optimization.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+```py
+import torch
+import time
+
+# Verify GPU selection else use CPU
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+# Heavy calculation function
+def heavy_computation(tensor):
+    return tensor**2 + (tensor**3) * (tensor.sin() * tensor.cos()) + tensor.tan()
```
Comment (avdhoottt) [inline content/pytorch/concepts/gpu-acceleration-with-cuda/terms/performance-optimization/performance-optimization.md:41]:
```suggestion
  return tensor**2 + (tensor**3) * (tensor.sin() * tensor.cos()) + tensor.tan()
```

File: content/pytorch/concepts/gpu-acceleration-with-cuda/terms/performance-optimization/performance-optimization.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+# Start time tracking for Stream computation
+start_time = time.time()
+
+# Asynchronous execution with streams
+with torch.cuda.stream(stream1):
+    result1 = heavy_computation(data[:size // 2])
+
+with torch.cuda.stream(stream2):
+    result2 = heavy_computation(data[size // 2:])
```
Comment (avdhoottt) [inline content/pytorch/concepts/gpu-acceleration-with-cuda/terms/performance-optimization/performance-optimization.md:62]:
```suggestion
with torch.cuda.stream(stream1):
  result1 = heavy_computation(data[:size // 2])

with torch.cuda.stream(stream2):
  result2 = heavy_computation(data[size // 2:])
```

**Review by avdhoottt — APPROVED** [review]
LGTM!!
